submissions/smillims/tiny-js-world/index.js

```diff
@@ -0,0 +1,50 @@
+/* 
+	Refer to https://github.com/OleksiyRudenko/a-tiny-JS-world for the task details
+   Complete the below for code reviewers' convenience:
+*/
+
+const cat = {
+	species: 'cat',
+	name: 'Murka',
+	gender: 'female',
+	legs: '4',
+	saying: 'Hey, I\'m cat.',
+};
+
+const man = {
+	species: 'human',
+	name: 'Jorj',
+	surname: 'Kluni',
+	gender: 'male',
+	legs: '2',
+	hands: '2',
+	saying:'Did you see Ocean\'s Eleven?',
+};
+
+const dog = {
+	species: 'dog',
+	name: 'Arrow',
+	gender: 'female',
+	legs: '4',
+	saying: 'I were in space, what about you?',
+};
+
+const woman = {
+	species: 'human',
+	name: 'Enn',
+	surname: 'Hatuey',
+	gender: 'female',
+	legs: '2',
+	hands: '2',
+	saying: 'Follow me on instagram.',
+}
+
+const all = [cat, man, dog, woman];
+
+const returnString = (arr) => {
+	return arr.map((person) => {
+		 return Object.values(person).join("; ");
```

Comment:
`Object` methods (`keys`, `values`, `entries`) as wel as `for...in` statement do not guarantee any particular order of keys/values. This means that under different JS engine the outcome might differ from expected.

Array guaranties the order. List properties or property names explicitly, map them into values (needed when list contains just names and not properties themselves) and then join.

Reply:
Well, you will have to resolve the case when an object has no specific property from those explicitly listed.

`Array.filter` and `Array.includes` are to your rescue. (`Object.keys` is fine for filtering purposes since order doesn't matter when checking if a key is a part of a particular object's key list).